lib/config.js

```diff
@@ -209,7 +209,7 @@ class Config {
             const localConfigHierarchyCache = this.configCache.getHierarchyLocalConfigs(localConfigDirectory);
 
             if (localConfigHierarchyCache) {
-                const localConfigHierarchy = localConfigHierarchyCache.concat(configs.reverse());
+                const localConfigHierarchy = localConfigHierarchyCache.concat(configs);
 
                 this.configCache.setHierarchyLocalConfigs(searched, localConfigHierarchy);
                 return localConfigHierarchy;
```

Comment:
I'm confused about this part of the change. Since `localConfigHierarchy` is returned here, it seems like this change will affect the behavior of the function, because localConfigHierarchy will have some configs in the wrong order. Is this intentional?

Reply:
I'm not sure it's wrong order -- I replaced all `configs.push(localConfig); ` => `configs.unshift(localConfig); `, so `configs` could always be same order as before. am I missing something?

Reply:
Since `configs` is used on line 212 instead of `configs.reverse()`, the configs will be stored in the cache in the opposite order from before.

This isn't necessarily a problem if this function is the only function that reads from that cache, but it could cause a problem if there are other parts of the code that read from the cache and expect the reverse order.

Reply:
oh, sorry for misleading. Actually `configs` would be always reverse order, so  the line 212 is the same as before `configs.reverse()`.


![image](https://user-images.githubusercontent.com/13050025/39952853-5777798c-55d2-11e8-83e2-4f2feea09b0b.png)